src/App.jsx

```diff
@@ -14,45 +16,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELDS = {
+  alphabet: 'alphabet',
+  length: 'length',
+};
+
+function getSortedGoods(sortField, isReversed = false) {
+  const preparedGoods = [...goodsFromServer];
+
+  if (sortField) {
+    preparedGoods.sort((goodA, goodB) => {
+      switch (sortField) {
+        case SORT_FIELDS.alphabet:
+          return goodA.localeCompare(goodB);
+
+        case SORT_FIELDS.length:
+          return goodA.length - goodB.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const [visibleGoods, setVisibleGoods] = useState([]);
+  const isShowResetButton = sortField || isReversed;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  useEffect(() => {
+    setVisibleGoods(getSortedGoods(sortField, isReversed));
+  }, [sortField, isReversed]);
+
+  const reset = () => {
```

Comment:
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364
```suggestion
  const handleButtonReset = () => {
```